homeassistant/components/yeelight/config_flow.py

```diff
@@ -0,0 +1,188 @@
+            if entry.data[CONF_ID]
+        }
+        devices_name = {}
+        # Run 3 times as packets can get lost
+        for _ in range(3):
+            devices = await self.hass.async_add_executor_job(yeelight.discover_bulbs)
+            for device in devices:
+                capabilities = device["capabilities"]
+                unique_id = capabilities["id"]
+                if unique_id in configured_devices:
+                    continue  # ignore configured devices
+                model = capabilities["model"]
+                ipaddr = device["ip"]
+                name = f"{ipaddr} {model} {unique_id}"
+                self._discovered_devices[unique_id] = capabilities
+                devices_name[unique_id] = name
+
+        if len(devices_name) == 0:
+            return self.async_abort(reason="no_devices_found")
+        return self.async_show_form(
+            step_id="pick_device",
+            data_schema=vol.Schema({vol.Required(CONF_DEVICE): vol.In(devices_name)}),
+        )
+
+    async def async_step_options(self, user_input=None):
+        """Handle the options step."""
+        if user_input is not None:
+            user_input[CONF_IP_ADDRESS] = self._ipaddr
+            user_input[CONF_ID] = self._unique_id
+            return self.async_create_entry(
+                title=user_input[CONF_NAME], data=user_input,
+            )
+
+        user_input = user_input or {}
+        return self.async_show_form(
+            step_id="options",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME, default=self._async_default_name()): str,
+                    vol.Optional(CONF_MODEL): str,
+                    vol.Required(
+                        CONF_TRANSITION, default=DEFAULT_TRANSITION
+                    ): cv.positive_int,
+                    vol.Required(CONF_MODE_MUSIC, default=DEFAULT_MODE_MUSIC): bool,
+                    vol.Required(
+                        CONF_SAVE_ON_CHANGE, default=DEFAULT_SAVE_ON_CHANGE
+                    ): bool,
+                    vol.Required(
+                        CONF_NIGHTLIGHT_SWITCH, default=DEFAULT_NIGHTLIGHT_SWITCH
+                    ): bool,
+                }
+            ),
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Handle import step."""
+        try:
+            await self._async_try_connect(user_input)
+        except CannotConnect:
+            _LOGGER.error("Failed to import %s: cannot connect", self._ipaddr)
+            return self.async_abort(reason="cannot_connect")
+        except AlreadyConfigured:
+            return self.async_abort(reason="already_configured")
+        if CONF_NIGHTLIGHT_SWITCH_TYPE in user_input:
+            user_input[CONF_NIGHTLIGHT_SWITCH] = (
+                user_input.pop(CONF_NIGHTLIGHT_SWITCH_TYPE)
+                == NIGHTLIGHT_SWITCH_TYPE_LIGHT
+            )
+        user_input[CONF_ID] = ""
```

Comment:
Why don't we use the id that we get from the capabilities?

Reply:
Manually added devices use IP addresses to connect. Should I just remove this from the entry?

Reply:
If we don't use, we can remove it.